pkg/sentry/fsimpl/fuse/init.go

```diff
@@ -0,0 +1,248 @@
+		if out.Minor >= 6 {
+			if out.Flags&linux.FUSE_ASYNC_READ != 0 {
+				fs.conn.AsyncRead = true
+			}
+
+			if out.Flags&linux.FUSE_POSIX_LOCKS == 0 {
+				fs.conn.NoLock = true
+			}
+
+			// No support for FLOCK flag before minor version 17.
+			if out.Minor >= 17 {
+				if out.Flags&linux.FUSE_FLOCK_LOCKS == 0 {
+					fs.conn.NoFLock = true
+				}
+			} else {
+				if out.Flags&linux.FUSE_POSIX_LOCKS == 0 {
+					fs.conn.NoFLock = true
+				}
+			}
+
+			if out.Flags&linux.FUSE_ATOMIC_O_TRUNC != 0 {
+				fs.conn.AtomicOTrunc = true
+			}
+
+			// No support for EXPORT flag before minor version 9.
+			if out.Minor >= 9 {
+				if out.Flags&linux.FUSE_EXPORT_SUPPORT != 0 {
+					fs.conn.ExportSupport = true
+				}
+			}
+
+			if out.Flags&linux.FUSE_BIG_WRITES != 0 {
+				fs.conn.BigWrites = true
+			}
+
+			if out.Flags&linux.FUSE_DONT_MASK != 0 {
+				fs.conn.DontMask = true
+			}
+
+			if out.Flags&linux.FUSE_AUTO_INVAL_DATA != 0 {
+				fs.conn.AutoInvalData = true
+			} else if out.Flags&linux.FUSE_EXPLICIT_INVAL_DATA != 0 {
+				fs.conn.ExplicitInvalData = true
+			}
+
+			if out.Flags&linux.FUSE_DO_READDIRPLUS != 0 {
+				fs.conn.DoReaddirplus = true
+				if out.Flags&linux.FUSE_READDIRPLUS_AUTO != 0 {
+					fs.conn.ReaddirplusAuto = true
+				}
+			}
+
+			if out.Flags&linux.FUSE_ASYNC_DIO != 0 {
+				fs.conn.AsyncDio = true
+			}
+
+			if out.Flags&linux.FUSE_WRITEBACK_CACHE != 0 {
+				fs.conn.WritebackCache = true
+			}
+
+			if out.Flags&linux.FUSE_PARALLEL_DIROPS != 0 {
+				fs.conn.ParallelDirops = true
+			}
+
+			if out.Flags&linux.FUSE_HANDLE_KILLPRIV != 0 {
+				fs.conn.HandleKillpriv = true
+			}
+
+			// if out.TimeGran > 0 && out.TimeGran <= fuseMaxTimeGranNs {
+			// TODO(gvisor.dev/issue/3195): figure out how to use this.
+			// superBlock.s_time_gran = reply.TimeGran
+			// }
+
+			if out.Flags&linux.FUSE_POSIX_ACL != 0 {
+				fs.conn.DefaultPermissions = true
+				fs.conn.PosixACL = true
+				// TODO(gvisor.dev/issue/3194): add xattr handler support.
+				// superBlock.xattrHandler =
+			}
+
+			if out.Flags&linux.FUSE_CACHE_SYMLINKS != 0 {
+				fs.conn.CacheSymlinks = true
+			}
+
+			if out.Flags&linux.FUSE_ABORT_ERROR != 0 {
+				fs.conn.AbortErr = true
+			}
+
+			if out.Flags&linux.FUSE_MAX_PAGES != 0 {
+				maxPages := out.MaxPages
+				if maxPages < 1 {
+					maxPages = 1
+				}
+				if maxPages > fuseMaxMaxPages {
+					maxPages = fuseMaxMaxPages
+				}
+				fs.conn.MaxPages = maxPages
+			}
+		} else {
+			fs.conn.NoLock = true
+			fs.conn.NoFLock = true
+		}
+
+		// No support for negotiating MaxWrite before minor version 5.
+		if out.Minor < 5 {
+			fs.conn.MaxWrite = fuseMinMaxWrite
+		} else {
+			fs.conn.MaxWrite = out.MaxWrite
+		}
+		if fs.conn.MaxWrite < fuseMinMaxWrite {
+			fs.conn.MaxWrite = fuseMinMaxWrite
+		}
+	}
+
+	fs.conn.setInitialized()
+
+	// TODO(gvisor.dev/issue/3185): unblock all blocked requests so far.
+	// close(fs.conn.blockedWaitQueueCh)
```

Comment:
I think you can implement this part. Here is an example how it can be done: https://github.com/avagin/gvisor/commit/9833f75d9037257b0b89d326a1ae75ca16678e7c

Reply:
* Not sure if one channel is enough. Each time there are `MaxBackground` requests, need to block, and reset afterwards
* Perhaps this PR is not a good place for that logic? The original intention was to wait for other PR to finish all the queuing logic (including but not limited to: congestion control, abort, block before init, block after MaxBackground, where to block (e.g. in `connection.Call` or another thread that handles sending all the requests from a single to-send request queue))